### PR TITLE
feat(developer): kmc emit hint if project file is old version 🦕

### LINKED ISSUE #10147: feat(developer): Add lint hint to kmc when compiling v1.0 project recommending upgrade
Note: this should probably honour the setting in #10146.

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #10147.

@keymanapp-test-bot skip

### CODE REVIEW

File: developer/src/kmc/src/commands/buildClasses/BuildProject.ts [hunk 1]
```diff
@@ -41,6 +41,11 @@ class ProjectBuilder {
       return false;
     }
 
+    // Give a hint if the project is v1.0
+    if(this.project.options.version != '2.0') {
```
Comment (darcywong00) [inline developer/src/kmc/src/commands/buildClasses/BuildProject.ts:45]: Wouldn't a version comparison (< 2.0) future-proof this check?

Comment (mcdurdin) [reply developer/src/kmc/src/commands/buildClasses/BuildProject.ts:45]: Perhaps, but we'll need to revisit all this anyway if we want to upgrade project file versions again, and we may want to do something different for that case.

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.228-alpha](https://keyman.com/downloads/releases/alpha/17.0.228)

**Pull request merged**